reason-react-navigation-example/src/TwitterNavigation.re

```diff
@@ -0,0 +1,628 @@
+        );
+      };
+    | None => route
+    }
+  );
+
+module Compose = {
+  [@react.component]
+  let make = (~navigation) => <View />;
+
+  make->setNavigationOptions(NavigationOptions.t(~title="Compose", ()));
+};
+
+module Profile = {
+  [@react.component]
+  let make = (~navigation) => placeholder(~navigation, ~text="Profile");
+
+  make->setNavigationOptions(NavigationOptions.t(~title="Profile", ()));
+};
+
+module Content = {
+  open DrawerNavigator;
+
+  module NavRow = {
+    [@react.component]
+    let make = (~navigation, ~route, ~title) =>
+      <TouchableOpacity
+        onPress={_ => {
+          navigation->Navigation.closeDrawer();
+          navigation->Navigation.navigate(route);
+        }}
+        style=Style.(style(~padding=14.->dp, ()))>
+        <Text style=Style.(style(~color="#444", ()))>
+          {title |> React.string}
+        </Text>
+      </TouchableOpacity>;
+  };
+
+  let contentComponent = (props: ContentComponent.props(unit)) => {
+    let navigation = props->ContentComponent.navigation;
+    <SafeAreaView style={Style.style(~flex=1., ())}>
+      <View
+        style=Style.(
+          style(~borderBottomWidth=1., ~borderBottomColor="#eee", ())
+        )>
+        <Image
+          style=Style.(
+            style(
+              ~backgroundColor="grey",
+              ~width=36.->dp,
+              ~borderRadius=18.,
+              ~margin=11.->dp,
+              ~height=36.->dp,
+              (),
+            )
+          )
+          source=Icons.avatar
+        />
+      </View>
+      <View
+        style=Style.(
+          style(~borderBottomWidth=1., ~borderBottomColor="#eee", ())
+        )>
+        <NavRow route="Profile" title="Profile" navigation />
+        <NavRow route="Lists" title="Lists" navigation />
+        <NavRow route="Bookmarks" title="Bookmarks" navigation />
+        <NavRow route="Moments" title="Moments" navigation />
+      </View>
+      <View
+        style=Style.(
+          style(~borderBottomWidth=1., ~borderBottomColor="#eee", ())
+        )>
+        <NavRow route="Settings" title="Settings and privacy" navigation />
+        <NavRow route="Help" title="Help Centre" navigation />
+      </View>
+    </SafeAreaView>;
+  };
+
+  let navigator =
+    makeWithConfig(
+      {
+        "Root":
+          TabNavigator.(
+            Bottom.makeWithConfig(
+              {
+                "Timeline": Timeline.navigator,
+                "Search": Search.navigator,
+                "Notifications": Notifications.navigator,
+                "Messages": Messages.navigator,
+              },
+              config(~initialRouteName="Timeline", ()),
+            )
+          ),
+      },
+      config(~contentComponent, ()),
+    );
+
+  navigator->setDynamicNavigationOptions(params => {
+    let navigation = params->NavigationParams.navigation;
+    let state = navigation->Navigation.state;
+    let routeName = getActiveRoute(state)->NavigationState.routeNameGet;
```

Comment:
We could simplify this patterns to (where `navigation` is retrieved & used once) via

```reason
let routeName = params
  ->NavigationParams.navigation
  ->Navigation.state
  ->getActiveRoute
  ->NavigationState.routeNameGet
```